### PR TITLE
Youth Financial Education: Fix flaky scroll into view tests

### PR DESCRIPTION
**Pull request by anselmbradford**
https://github.com/cfpb/consumerfinance.gov/pull/7421 made some flaky tests run headless, but they still acted flaky. This test replaces the waits in those tests to check that an element at the top or bottom of the page has scrolled into view.

EDIT: looks like we still need the waits :-\

## Additions

- `isScrolledTo` Cypress command.

## Changes

- ~Replace arbitrary waits with checking~ Check if an element at the top or bottom of the page has been scrolled into view.

## How to test this PR

1. PR functional test checks should pass.

### CODE REVIEW

File: test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js [hunk 1]
```diff
@@ -2,20 +2,21 @@ import { TdpSurveyHelpers } from './survey-helpers.cy.js';
...
+}
 
+describe('Youth Financial Education Survey: Errors', () => {
   it('jumps to errors at top', () => {
     refreshErrors();
     cy.wait(1200);
+    cy.get('main h1').isScrolledTo();
     cy.window().then((win) => {
       expect(win.scrollY).lessThan(400);
     });
```
Comment (niqjohnson) [inline test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js:22]: Now that we're checking to see if a certain element is in view, I wonder if the check for the correct `scrollY` coordinates is redundant? From what I can tell, the intention of the test is "make sure the right part of the page is visible after taking this action"; seems like that would be satisfied by checking that all the right elements pass the `isScrolledTo` assertion. I might be misunderstanding what's actually being checked there, though.

File: test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js [hunk 2]
```diff
@@ -51,6 +52,7 @@ describe('Youth Financial Education Survey: Errors', () => {
     refreshErrors();
     cy.get('form .m-notification__error li:nth-child(2) a').click();
     cy.wait(1200);
+    cy.get('.survey-reset--link--wrap').isScrolledTo();
     cy.window().then((win) => {
       expect(win.scrollY).greaterThan(1000);
     });
```
Comment (niqjohnson) [inline test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js:58]: Now that we're checking to see if a certain element is in view, I wonder if the check for the correct `scrollY` coordinates is redundant? From what I can tell, the intention of the test is "make sure the right part of the page is visible after taking this action"; seems like that would be satisfied by checking that all the right elements pass the `isScrolledTo` assertion. I might be misunderstanding what's actually being checked there, though.

**Review by niqjohnson — COMMENTED** [review]
Same comments in two spots: I wonder if we can ditch the checks on `scrollY` as redundant now that we have the `isScrolledTo` checks.

File: test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js (hunk 1 above)
Comment (anselmbradford) [reply test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js]: Works for me, I think you are right. Changed in https://github.com/cfpb/consumerfinance.gov/pull/7424/commits/597c32f8ade337145d4f10b4b0cc9eab3d0475f2

File: test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js (hunk 2 above)
Comment (anselmbradford) [reply test/cypress/integration/consumer-tools/youth-financial-education/survey-errors.cy.js]: Works for me, I think you are right. Changed in https://github.com/cfpb/consumerfinance.gov/pull/7424/commits/597c32f8ade337145d4f10b4b0cc9eab3d0475f2

**Comment by anselmbradford** [comment]
Testing removing the waits again. I think with the `isScrolledTo` command only they may not be needed.

**Comment by anselmbradford** [comment]
Test appears to be a success!

**Review by wpears — APPROVED** [review]

**Pull request merged**